### ISSUE TITLE
features.json is missing from Windows installer for clang and docc

### ISSUE THREAD

**Issue by jakepetroules** [swiftlang/swift-installer-scripts#337]
We probably need a section similar to this, but for clang and docc:

    <ComponentGroup Id="SwiftFeatures" Directory="_usr_share_swift">
      <Component>
        <File Source="$(TOOLCHAIN_ROOT)\usr\share\swift\features.json" />
      </Component>
    </ComponentGroup>

Note that clang does have a ComponentGroup for its features.json (and I don't see any for docc), but only swift's is actually present in the installer.

**Comment by compnerd** [comment]
DocC currently isn't usable on windows due to the missing pending change (CC: @FranzBusch). I'm happy to add the features.json for clang though.

**Comment by compnerd** [comment]
https://github.com/swiftlang/swift-installer-scripts/blob/main/platforms/Windows/bld/bld.wxs#L158-L162

It seems that the clang features.json is already present. @jakepetroules is there something that I'm missing here?

**Comment by jakepetroules** [comment]
It's referenced in the WiX sources but doesn't actually get installed on the system. Does WiX not error out if it references a file that doesn't exist when building the installer?

**Comment by compnerd** [comment]
Yes, if the file is missing the installer will fail to build.

**Comment by jakepetroules** [comment]
Oh, looks like you fixed clang in fe35b366c51c4d9d5ebeefe35ea5bc49f7c6925d, but that's not in the 6.0 branch. I'll update the title to reflect that this is only an issue for docc, then.

What's the docc usability issue you were referencing above?

**Comment by compnerd** [comment]
DocC cannot generate the documentation on Windows due to the use of `:` in the filenames. There is an existing PR (https://github.com/swiftlang/swift-docc/pull/668) that adjusts that, but I think that there is still lack of clarity on how to migrate to that.

**Comment by jakepetroules** [comment]
Thanks for the reference. I think those two issues could be resolved in parallel, though.

**Comment by compnerd** [comment]
Yeah, I suppose that we could start building it. The one issue is the build times - I think that using CMake resulted in a 7% build time improvement as well as alleviated some bloat due to being able to do dynamic linking against other toolchain components.

**Comment by jakepetroules** [comment]
To be clear, we _are_ currently shipping docc.exe as part of the Swift Windows installer already. Just the features.json is missing.

**Comment by compnerd** [comment]
> To be clear, we _are_ currently shipping docc.exe as part of the Swift Windows installer already. Just the features.json is missing.

Interesting, I don't remember getting that enabled, but that is great!

**Comment by jakepetroules** [comment]
Oh, you may be right. I remember seeing it in _some_ toolchain build at least, maybe that was an older version and it's since been removed again.
